### PR TITLE
Task contribution

### PR DESCRIPTION
**Pull request by anthonydresser**
Adds a task contribution for running contributed commands as tasks. The package.json uses the command contribution that vscode uses, but when you register the command in the extension you instead call sqlops.tasks.registerTask. The callback will be provided the IConnectionProfile that is the context of the task being executed. Additionally we recognized these commands internally as tasks so they can be used accordingly, for example in the tasks widget on the dashboard.

I exposed the IConnectionProfile object in the typings for use in the extensions because a lot of the internal tasks we use need this profile, but we didn't have a great way to get it from currently exposed methods.

### CODE REVIEW

File: src/sql/workbench/api/node/extHostTasks.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		try {
+			let result = callback.apply(thisArg, args);
+			return TPromise.as(result);
+		} catch (err) {
+			// console.log(err);
+			// try {
+			// 	console.log(toErrorMessage(err));
+			// } catch (err) {
+			// 	//
+			// }
```
Comment (MattIrv) [inline src/sql/workbench/api/node/extHostTasks.ts]: commented code

**Comment by kevcunnane** [comment]
👍 to Matt's comment on a unified representation of common items like the connection profile.

**Comment by anthonydresser** [comment]
@MattIrv the problem with having a method off connection management service, is sometimes we change the connection profiles without making any changes in the connection itself. for example the profile for a database on OE adds the database name to the connection profile when calling actions, but it doesn't change anything in connection management service. This is also true in dashboard. So in order to implement what you are suggesting would require a lot of changes throughout the product to make sure everything is synced with the connection management service.

**Comment by MattIrv** [comment]
@anthonydresser got it, your call on what the best way forward is then. I'm not a fan of having connections managed/changed when those changes are not visible to the connection management service, but if that's how we already do things then no need to change it now

**Pull request merged**